src/ruby/ext/grpc/rb_server.c

```diff
@@ -310,6 +314,29 @@ static VALUE grpc_rb_server_add_http2_port(VALUE self, VALUE port,
   return INT2NUM(recvd_port);
 }
 
+static VALUE grpc_rb_server_add_insecure_channel_from_fd(VALUE self, VALUE rb_fd) {
+#ifdef GPR_SUPPORT_CHANNELS_FROM_FD
```

Comment:
same issue in this constructor as with the client channel. the `ifdef` could be remove here and a ruby wrapper for `grpc_server_add_insecure_channel_from_fd` could wrap a smaller amount of code

Reply:
same as above comment, can we please move the code inside the ifdef into its own function, and then only call that function inside the ifdef?